spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/notify/MicrosoftTeamsNotifier.java

```diff
@@ -0,0 +1,372 @@
+        this.deRegisteredTitle = deRegisteredTitle;
+    }
+
+    public String getRegisteredTitle() {
+        return registeredTitle;
+    }
+
+    public void setRegisteredTitle(String registeredTitle) {
+        this.registeredTitle = registeredTitle;
+    }
+
+    public String getStatusChangedTitle() {
+        return statusChangedTitle;
+    }
+
+    public void setStatusChangedTitle(String statusChangedTitle) {
+        this.statusChangedTitle = statusChangedTitle;
+    }
+
+    public String getMessageSummary() {
+        return messageSummary;
+    }
+
+    public void setMessageSummary(String messageSummary) {
+        this.messageSummary = messageSummary;
+    }
+
+
+    @JsonSerialize
+    class Message {
+
+        private String summary;
+        private String themeColor;
+        private String title;
+        private List<Section> sections = new ArrayList<>();
+
+        public String getSummary() {
+            return summary;
+        }
+
+        public void setSummary(String summary) {
+            this.summary = summary;
+        }
+
+        public String getTitle() {
+            return title;
+        }
+
+        public String getThemeColor() {
+            return themeColor;
+        }
+
+        public void setSections(List<Section> sections) {
+            this.sections = sections;
+        }
+
+        public void setTitle(String title) {
+            this.title = title;
+        }
+
+        public void setThemeColor(String themeColor) {
+            this.themeColor = themeColor;
+        }
+
+        public List<Section> getSections() {
+            return sections;
+        }
+
+    }
+
+    @JsonSerialize
+    class Section {
+
+        private String activityTitle;
+        private String activitySubtitle;
+        private String activityImage;
+        private List<Fact> facts = new ArrayList<Fact>();
+        private String text;
+
+        public String getActivityTitle() {
+            return activityTitle;
+        }
+
+        public void setActivityTitle(String activityTitle) {
+            this.activityTitle = activityTitle;
+        }
+
+        public String getActivitySubtitle() {
+            return activitySubtitle;
+        }
+
+        public void setActivitySubtitle(String activitySubtitle) {
+            this.activitySubtitle = activitySubtitle;
+        }
+
+        public void setFacts(List<Fact> facts) {
+            this.facts = facts;
+        }
+
+        public String getText() {
+            return text;
+        }
+
+        public void setText(String text) {
+            this.text = text;
+        }
+
+        public String getActivityImage() {
+            return activityImage;
+        }
+
+        public void setActivityImage(String activityImage) {
+            this.activityImage = activityImage;
+        }
+
+        public List<Fact> getFacts() {
+            return facts;
+        }
+    }
+
+    @JsonSerialize
+    class Fact {
```

Comment:
either don't make it an inner class or make it static

Reply:
Addressed in [95d0b934cb83ed50f29baefc70f583f901db4c87](https://github.com/codecentric/spring-boot-admin/pull/592/commits/95d0b934cb83ed50f29baefc70f583f901db4c87)